src/compiler/checker.ts

```diff
@@ -6254,16 +6254,16 @@ namespace ts {
                 const parameterName = node.parameterName as Identifier;
                 return {
                     kind: TypePredicateKind.Identifier,
-                    parameterName: parameterName ? parameterName.text : undefined,
+                    parameterName: parameterName ? <string>parameterName.text : undefined,

```

Comment:
This cast is unsafe and probably indicative of a bug somewhere. Something to do with underscore prefixed identifiers and type guards. Use `unescapeLeadingUnderscores(parameterName.text)` instead.